src/main/kotlin/org/jitsi/nlj/RtpLayerDesc.kt

```diff
@@ -123,16 +152,33 @@ constructor(
      * and its dependencies.
      */
     fun getBitrate(nowMs: Long): Bandwidth {
-        var bitrate = bitrateTracker.getRate(nowMs)
+        val rates = HashMap<Int, Bandwidth>()
+
+        getBitrate(nowMs, rates)
+
+        return rates.values.sum()
+    }
 
-        /* TODO: does the wrong thing if we have multiple dependencies */
-        if (dependencyLayers != null) {
-            for (dependency in dependencyLayers) {
-                bitrate += dependency.getBitrate(nowMs)
-            }
+    /**
+     * Recursively adds the bitrate (in bps) of this [RTPLayerDesc] and
+     * its dependencies in the map passed in as an argument.
+     *
+     * This is necessary to ensure we don't double-count layers in cases
+     * of multiple dependencies.
+     *
+     * @param nowMs
+     */
+    private fun getBitrate(nowMs: Long, rates: MutableMap<Int, Bandwidth>) {
```

Comment:
I don't follow what you have in mind here

Reply:
If this function were 

```
private fun getBitrate(nowMs: Long, rates: MutableMap<Int, Bandwidth>) = HashMap()) { 
```

Then, when calling it the initial time, you could avoid having to declare your own map and pass it in.  So instead of:
```
val map = HashMap<Int, BandWidth>()
val rate = getBitrate(nowMs, map)
```

It'd just be

```
val rate = getBitrate(nowMs)
```

The recursive calls would still pass in their map to carry things over.

Reply:
The problem is that `getBitrate(nowMs)` is the signature of the public function.  I suppose I could call the internal function something different though.

Reply:
I've done that, see what you think.